spec/lib/dor/text_extraction/abbyy/results_spec.rb

```diff
@@ -28,11 +35,26 @@
     it 'does have an alto doc' do
       expect(results.alto_doc).to eq '/tmp/OUTPUT/bb222cc3333/bb222cc3333.xml'
     end
+
+    it 'records the software name and version' do
+      expect(results.software_name).to eq 'ABBYY FineReader Server'
+      expect(results.software_version).to eq '14.0'
+    end
+
+    it 'stores the last seen software name and version' do
+      expect(described_class.last_software_name).to eq 'ABBYY FineReader Server'
+      expect(described_class.last_software_version).to eq '14.0'
+    end
```

Comment:
i think having this in its own `it` block introduces a touch of order dependence, though i had to do something pretty contrived to trigger it, so maybe this will be so minimally flaky (or maybe my tweaks were contrived enough) that this won't be an issue in practice.

but if i comment out all the tests besides the two related to `last_software_name` and `last_software_version`, and run this spec file, i get intermittent failures (my first try at that used seed 41393 and failed, but my couple runs after that both passed -- for the passing runs, the test at line 74 and its `before` block ran first according to rspec output).

anyway, i wonder if you'd consider either: combining the `it 'records the software name and version'` and `it 'stores the last seen software name and version'` expectations into one `it` (which of course would lead to slightly less informative test failures since the expectations in the block would be slightly more disparate); or putting this `it` in its own `context` with the same  sort of class var setting `before` as the `context 'when results do not render'`.  i think i like the first option more, just because it exercises the actual caching mechanism a little more realistically, which seems closer to the intent of the current test (and that feels like a more worthwhile test to me).

Reply:
yeah, the tests expose just how janky this strategy is, lol. simplest thing to do is combine them, I think. i'll do that